### ISSUE TITLE
Optimizer test failing for theano

### ISSUE THREAD

**Issue by achalshah20** [fchollet/keras#5499]
Why we are checking `assert history.history['acc'][-1] >= target` in this below code.

in tests/keras/optimizers_test.py

def _test_optimizer(optimizer, target=0.75):
    x_train, y_train = get_test_data()
    model = get_model(x_train.shape[1], 10, y_train.shape[1])
    model.compile(loss='categorical_crossentropy',
                  optimizer=optimizer,
                  metrics=['accuracy'])
    history = model.fit(x_train, y_train, epochs=5, batch_size=16, verbose=0)
    **assert history.history['acc'][-1] >= target**
    config = optimizers.serialize(optimizer)
    optim = optimizers.deserialize(config)
    new_config = optimizers.serialize(optim)
    new_config['class_name'] = new_config['class_name'].lower()
    assert config == new_config

With latest theano, this check always fails and this check doesn't make sense to me. Because in the first iteration it may not go above 0.75. But if I change epoch to 3, all test cases are passed (more than ~0.90). 

Can someone explain the motivation behind this check?

**Comment by the-moliver** [comment]
I find the same thing. Since it passes with TF it means they are behaving differently, which needs more investigation.

**Comment by the-moliver** [comment]
This was caused by different initial weights under Theano and Tensorflow. When testing with identical starting weights, the optimizers perform identically (within precision error) as they should. So relaxing the tests as here seems like the right move: https://github.com/fchollet/keras/commit/4b5984b8ce8799601ddf8f3fae0344b626c4ba48

**Comment by fchollet** [comment]
One remaining issue (both in TF and Theano) is that the initial learning
rate of Adadelta is pretty bad currently. I found that using a higher
learning rate works better.

**Comment by the-moliver** [comment]
Huh, that's interesting since the original paper didn't even have a learning rate parameter. Does it seem to universally work better, and do you have to decay it?
